data/templates/esx-ks

```diff
@@ -8,24 +8,22 @@ accepteula
 <% } %>
 rootpw <%=rootPlainPassword%>
 
-# Search the networkDevices and set the first vmnic# (if defined) up.
-# If no vmnic# device is specified in the networkDevices, then we fallback
-# to setting vmnic0 up as dhcp
+# Search the networkDevices and set the first device (if defined) up.
+# If no device is specified in the networkDevices, then we fallback
+# to setting 'vmnic0' up as DHCP. The device can be specified with a
+# MAC address or device name ('vmnic0' for example)
 <% if( typeof networkDevices !== 'undefined' ) { %>
   <% need_default = networkDevices.every(function(n) { %>
-    <% if( n.device.substring(0,5) === 'vmnic' ) { %>
-      <% if(typeof n.ipv4 !== 'undefined') { %>
-        <% ipopts = '--ip=' + n.ipv4.ipAddr + ' --gateway=' + n.ipv4.gateway + ' --netmask=' + n.ipv4.netmask %>
-        <% if (typeof n.ipv4.vlanIds !== 'undefined' ) { %>
-          <% ipopts += ' --vlandid=' + n.ipv4.vlanIds[0] %>
-        <% } %>
-        network --bootproto=static --device=<%=n.device%> <%=ipopts%>
-      <% } else { %>
-        network --bootproto=dhcp --device=<%=n.device%>
+    <% if(typeof n.ipv4 !== 'undefined') { %>
+      <% ipopts = '--ip=' + n.ipv4.ipAddr + ' --gateway=' + n.ipv4.gateway + ' --netmask=' + n.ipv4.netmask %>
+      <% if (typeof n.ipv4.vlanIds !== 'undefined' ) { %>
+        <% ipopts += ' --vlandid=' + n.ipv4.vlanIds[0] %>
       <% } %>
-      <% return false; %>
+      network --bootproto=static --device=<%=n.device%> <%=ipopts%>
+    <% } else { %>
+      network --bootproto=dhcp --device=<%=n.device%>
     <% } %>
-    <% return true; %>
+    <% return false; %>
```

Comment:
In the original template the 'device' name maps to the portgroup name and does not have to be the physical interface used to boot.  Returning 'false' here breaks that and forces the first interface to be the physical interface that is being used to load ESXi.  This new behavior should probably be documented somewhere.